drivers/flash/flash_stm32_ospi.c

```diff
@@ -0,0 +1,1498 @@
+				? HAL_OSPI_ADDRESS_1_LINE
+				: HAL_OSPI_ADDRESS_8_LINES;
+	s_command.Address        = 0U;
+	s_command.DummyCycles    = (nor_mode == OSPI_SPI_MODE)
+				? 0U
+				: ((nor_rate == OSPI_DTR_TRANSFER)
+				? SPI_NOR_DUMMY_REG_OCTAL_DTR
+				: SPI_NOR_DUMMY_REG_OCTAL);
+	s_command.DataMode       = (nor_mode == OSPI_SPI_MODE)
+				? HAL_OSPI_DATA_1_LINE
+				: HAL_OSPI_DATA_8_LINES;
+	s_command.NbData         = (nor_rate == OSPI_DTR_TRANSFER) ? 2U : 1U;
+
+	/* Send the command */
+	if (HAL_OSPI_Command(hospi, &s_command, HAL_OSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
+		LOG_ERR("OSPI config auto polling cmd failed");
+		return -EIO;
+	}
+
+	s_config.Match           = SPI_NOR_WREN_MATCH;
+	s_config.Mask            = SPI_NOR_WREN_MASK;
+	s_config.MatchMode       = HAL_OSPI_MATCH_MODE_AND;
+	s_config.Interval        = SPI_NOR_AUTO_POLLING_INTERVAL;
+	s_config.AutomaticStop   = HAL_OSPI_AUTOMATIC_STOP_ENABLE;
+
+	if (HAL_OSPI_AutoPolling(hospi, &s_config, HAL_OSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
+		LOG_ERR("OSPI config auto polling failed");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+/* Write Flash configuration register 2 with new dummy cycles */
+static int stm32_ospi_write_cfg2reg_dummy(OSPI_HandleTypeDef *hospi,
+					uint8_t nor_mode, uint8_t nor_rate)
+{
+	OSPI_RegularCmdTypeDef s_command = s_command = ospi_prepare_cmd(nor_mode, nor_rate);
+
+	/* Initialize the writing of configuration register 2 */
+	s_command.Instruction = (nor_mode == OSPI_SPI_MODE)
+				? SPI_NOR_CMD_WR_CFGREG2
+				: SPI_NOR_OCMD_WR_CFGREG2;
+	s_command.Address = SPI_NOR_REG2_ADDR3;
+	s_command.DummyCycles = 0U;
+	s_command.NbData = (nor_mode == OSPI_SPI_MODE) ? 1U
+			: ((nor_rate == OSPI_DTR_TRANSFER) ? 2U : 1U);
+
+	/* Send the command */
+	if (HAL_OSPI_Command(hospi, &s_command, HAL_OSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
+		LOG_ERR("OSPI transmit ");
+		return -EIO;
+	}
+	uint8_t tmp = SPI_NOR_CR2_DUMMY_CYCLES_66MHZ;
+
+	if (HAL_OSPI_Transmit(hospi, &tmp, HAL_OSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
+		LOG_ERR("OSPI transmit ");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+/* Write Flash configuration register 2 with new single or octal SPI protocol */
+static int stm32_ospi_write_cfg2reg_io(OSPI_HandleTypeDef *hospi,
+				       uint8_t nor_mode, uint8_t nor_rate, uint8_t op_enable)
+{
+	OSPI_RegularCmdTypeDef s_command = s_command = ospi_prepare_cmd(nor_mode, nor_rate);
+
+	/* Initialize the writing of configuration register 2 */
+	s_command.Instruction = (nor_mode == OSPI_SPI_MODE)
+				? SPI_NOR_CMD_WR_CFGREG2
+				: SPI_NOR_OCMD_WR_CFGREG2;
+	s_command.Address = SPI_NOR_REG2_ADDR1;
+	s_command.DummyCycles = 0U;
+	s_command.NbData = (nor_mode == OSPI_SPI_MODE) ? 1U
+			: ((nor_rate == OSPI_DTR_TRANSFER) ? 2U : 1U);
+
+	/* Send the command */
+	if (HAL_OSPI_Command(hospi, &s_command, HAL_OSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
+		LOG_ERR("Write Flash configuration reg2 failed");
+		return -EIO;
+	}
+	uint8_t tmp = op_enable;
```

Comment:
> uint8_t tmp = op_enable;

Could that be moved to the top of the function and named in a way to add some meaning?

Reply:
`HAL_OSPI_Transmit(hospi, &op_enable` is even better